### ISSUE TITLE
Expand guide_colourbar() tick options

### ISSUE THREAD

**Issue by wpetry** [tidyverse/ggplot2#1994]
Enable 'tick' argument to take additional arguments to control colour and size as in element_line().  For example, the default tick color is incompatible with colour bars containing light greys:
```r
data("faithfuld")
ggplot(faithfuld, aes(waiting, eruptions)) +
       geom_raster(aes(fill = density)) +
       scale_fill_gradient(low = "white", high = "grey70")
```

**Comment by hadley** [comment]
Yes, this really should be a theme setting.

**Comment by wpetry** [comment]
I agree with @thomasp85 that enabling these settings in `guide_colourbar()` would make a bit more sense than putting them in `theme()` because ticks aren't common in other legend types. 

Currently, the only setting to control ticks in the colour bar reside in `guide_colourbar()`. Perhaps the most consistent/least invasive solution would be to expand the `ticks` argument beyond a boolean? Something like `guide_colourbar(ticks = element_line(...))`? Here `element_line()` could be populated with arguments to match the current behaviour of the colour bar (i.e., `size = {something thin}, colour = "white"`). Users who desire to suppress ticks entirely would simply use `guide_colourbar(ticks = element_blank())`.

This gets a bit confusing because the colour options for the legend key border are in `theme(legend.key = element_rect(...))`. For example, a user wanting a black border around their colour bar _**and**_ black ticks in the colour bar would control these in two different functions.

**Comment by hadley** [comment]
Fixed in #2541
